File: Competetive programming/C++/Unit 2/Hartal.cpp
#include <iostream>
using namespace std;

int main() {

    int days, parties;

    cout << "Enter total number of days: ";
    cin >> days;

    cout << "Enter number of political parties: ";
    cin >> parties;

    int h[20];

    cout << "Enter hartal parameter for each party:\n";

    for(int i = 0; i < parties; i++)
        cin >> h[i];

    int lostDays = 0;

    for(int day = 1; day <= days; day++) {

        if(day % 7 == 6 || day % 7 == 0)
            continue;

        for(int i = 0; i < parties; i++) {

            if(day % h[i] == 0) {
                lostDays++;
                break;
            }
        }
    }

    cout << "\nLost Working Days = " << lostDays;

    return 0;
}

File: Competetive programming/Python/Unit 2/Hartal.py
print("===== HARTALS =====\n")

days = int(input("Enter total number of days: "))

parties = int(input("Enter number of political parties: "))

h = []

print("Enter hartal parameter for each party:")

for i in range(parties):
    h.append(int(input()))

lost_days = 0

for day in range(1, days + 1):

    if day % 7 == 6 or day % 7 == 0:
        continue

    for i in range(parties):

        if day % h[i] == 0:
            lost_days += 1
            break

print("\nLost Working Days =", lost_days)